modules/ROOT/pages/sysconfig-network-configuration.adoc

```diff
@@ -0,0 +1,966 @@
+either configured by the user or by the platform, during the initramfs.
+This is primarily done right now for VMWare and allows for users to
+specify networking configuration in the form of dracut networking
+arguments without having to stop the boot of the machine and manually
+inject those arguemnts themselves.
+
+See 
+https://github.com/coreos/afterburn/blob/master/docs/usage/initrd-network-cmdline.md[the Afterburn documentation]
+for more information.
+
+==== via Ignition
+
+Networking configuration can be performed by writing out files
+described in an Ignition config. These are 
+https://developer.gnome.org/NetworkManager/stable/nm-settings-keyfile.html[NetworkManager keyfiles]
+that are written to `/etc/NetworkManager/system-connections/` that
+tell NetworkManager what to do.
+
+Any configuration provided via Ignition will generally take precedence
+over any other method of configuring the Network for a Fedora CoreOS
+instance.
+
+An example FCCT configuration for the same static networking example
+that we showed above is:
+
+[source, yaml]
+----
+variant: fcos
+version: 1.0.0
+storage:
+  files:
+    - path: /etc/NetworkManager/system-connections/ens2.nmconnection
+      mode: 0600
+      contents:
+        inline: |
+          [connection]
+          id=ens2
+          type=ethernet
+          interface-name=ens2
+          [ipv4]
+          address1=10.10.10.10/24,10.10.10.1
+          dns=8.8.8.8;
+          dns-search=
+          may-fail=false
+          method=manual
+----
+
+
+== Host Network Configuration Examples
+
+In this section we'll go through common examples of setting up
+different types of networking devices using both dracut kernel
+arguments as well as NetworkManager keyfiles via Ignition/FCCT.
+
+Examples in this section that use a static IP will assume these
+values unless otherwise stated:
+
+[source, bash]
+----
+ip='10.10.10.10'
+gateway='10.10.10.1'
+netmask='255.255.255.0'
+prefix='24'
+hostname='myhostname'
+interface='ens2'
+nameserver='8.8.8.8'
+bondname='bond0'
+teamname='team0'
+bridgename='br0'
+subnic1='ens2'
+subnic2='ens3'
+vlanid='100'
+----
+
+NOTE: FCOS uses https://www.freedesktop.org/wiki/Software/systemd/PredictableNetworkInterfaceNames/[predictable interface names]
+      by https://lists.fedoraproject.org/archives/list/coreos-status@lists.fedoraproject.org/thread/6IPTZL57Z5NLBMPYMXNVSYAGLRFZBLIP/[default].
+      Please take care to use the correct interface name for your hardware.
+
+=== Generating NetworkManager Keyfiles using `nm-initrd-generator`
+
+NetworkManager ships a tool,
+https://developer.gnome.org/NetworkManager/stable/nm-initrd-generator.html[nm-initrd-generator],
+that can generate keyfiles from dracut kernel argument syntax. This
+might be a good way to either convert from kernel arguments to keyfiles
+or to just quickly generate some keyfiles giving a small amount of input
+and then tweak some more detailed settings.
```

Comment:
Can we show an example of running `nm-initrd-generator` in a container?

Reply:
I'd rather not. I don't know of any containers with NetworkManager installed in them.